### PR TITLE
feat: Update header layout following designs

### PR DESCRIPTION
**Pull request by winter-sols**
issue: #446 
Updated header following Figma design

### CODE REVIEW

**Comment by winter-sols** [comment]
@predrag-us, @zolotokrylin , @georgeciubotaru , @markholdex 
Would you like to this PR, please?

**Comment by winter-sols** [comment]
@predrag-us I can't understand your last 3 indications.
Could you help me with more description?

**Comment by winter-sols** [comment]
@predrag-us could you explain your notes? so I can understand clearly

**Comment by predrag-us** [comment]
@winter-sols 

> The secondary navigation should be hidden and appear on scroll (mind the notes)

[Click me. ](https://www.figma.com/file/39MGGVIGQ1bL3FgVzuXEdC/Holdex?type=design&node-id=3431%3A52992&mode=design&t=ulxlZMiqcg58w93x-1)

> The active indicator in the secondary navigation has the wrong position (should be above the line)

Implementation:

CleanShot 2023-07-19 at 17 14 56@2x

Design: 

CleanShot 2023-07-19 at 17 15 18@2x

> The `fade` is missing on the secondary navigation on mobile screens

Notice the `fade` that indicates that there's more on scroll. You're missing the `fade` in your implementation, as well as the option to scroll the menu.

CleanShot 2023-07-19 at 17 15 59@2x

**Comment by winter-sols** [comment]
@predrag-us Could you review again?
FYI, to scroll second navigation horizontally, press and hold the Shift while scrolling up (scroll left) and down (scroll right) with the mouse scroll wheel.

**Comment by predrag-us** [comment]
A new preview is required. 
cc: @georgeciubotaru

**Comment by markholdex** [comment]
@predrag-us new preview deployed. Thx!

**Comment by predrag-us** [comment]
@winter-sols 

We're going in a good direction, but there're some issues that require resolving:

- Holdex has a logo. Please check the footer and use the same logo in the navigation. 
- Secondary navigation is glitching when scrolling, please check on that. Also, wherever you are on the page, on scroll up the navigation should reveal. Feel free to apply a subtle fade animation. 
- Notice the doubled border once the secondary navigation is hidden.
- You're missing the main CTA on small devices. 
- You're missing fade on the left, which should appear once you scroll to the right. 
- Likewise, the fade on the right should disappear once you scrolled to the end of the secondary navigation. 
- Notice that your fade is affecting the borders at the top and bottom, that shouldn't be the case. 

Hope this helps.

**Comment by winter-sols** [comment]
@predrag-us 
`Secondary navigation is glitching when scrolling, please check on that. Also, wherever you are on the page, on scroll up the navigation should reveal. Feel free to apply a subtle fade animation.`
and also this
`You're missing the main CTA on small devices.`
Could you give me more description above two problems?

**Comment by winter-sols** [comment]
@predrag-us I'm waiting for your answer

**Comment by predrag-us** [comment]
Updated preview required.
cc: @markholdex @georgeciubotaru

File: src/routes/(pages)/+layout.svelte [hunk 1]
```diff
@@ -121,3 +121,13 @@
...
+
+<style lang="sass">
+  .scrollbar-hide::-webkit-scrollbar
+    display: none
+
+  /* For IE, Edge and Firefox */
+  .scrollbar-hide 
+    -ms-overflow-style: none  /* IE and Edge */
+    scrollbar-width: none  /* Firefox */
+</style>
```
Comment (georgeciubotaru) [inline src/routes/(pages)/+layout.svelte:133]: @winter-sols Why do we need that?

**Comment by predrag-us** [comment]
@winter-sols 

- Please notice the logo size, it's not the same as in the design
- The size of the logo in the footer is also not according to the design. Please fix this one as well along the way. Thanks!

---

> You're missing the main CTA on small devices.

If you have a look at the design, you'll notice that at some point we're removing `Partner With Us` from the navigation and include a fixed button in the bottom-right corner instead. 

[Here's a URL ](https://www.figma.com/file/39MGGVIGQ1bL3FgVzuXEdC/Holdex?type=design&node-id=1545%3A29477&mode=design&t=3V7Ohg8A9UdoX3aU-1)to the component. Mind the note on the right. 

Here's a preview: 

CleanShot 2023-07-21 at 09 25 33@2x

---

> Secondary navigation is glitching when scrolling, please check on that. Also, wherever you are on the page, on scroll up the navigation should reveal. Feel free to apply a subtle fade animation.

When we scroll down, the navigation hides, and you did that part correctly.
What we're missing is navigation showing when we scroll up.
So, no matter where you are on the page, scroll up should reveal the navigation.

Bear in mind that there are some glitching (content moving) when the navigation is hidden, and it will probably happen as well when the navigation shows on scroll up. Make sure to remove that glitching. 

---

You haven't completely solved the fade problem. 

Here're my comments once again: 

- You're missing fade on the left, which should appear once you scroll to the right.
- Likewise, the fade on the right should disappear once you scrolled to the end of the secondary navigation.

Also, I'm noticing this weird moment: 

CleanShot 2023-07-21 at 09 35 55@2x

File: src/routes/(pages)/+layout.svelte (hunk 1 above)
Comment (winter-sols) [reply src/routes/(pages)/+layout.svelte:133]: @georgeciubotaru This is needed to hide horizontal scroll bar on secondary navigation.

**Comment by winter-sols** [comment]
@predrag-us , @georgeciubotaru I addressed all problems.
Please review this again.

**Comment by predrag-us** [comment]
Updated preview required.
cc: @markholdex @georgeciubotaru

**Comment by winter-sols** [comment]
@georgeciubotaru @predrag-us @markholdex 
Could you give me updates?

**Comment by predrag-us** [comment]
@winter-sols 

- The fade is there when scrolling to the very end but not there when I'm at around 99%
- Check the design of the CTA, it's different based on the color modes
- Check the position of the CTA, doesn't seem right

**Comment by winter-sols** [comment]
@predrag-us I moved the CTA button a bit. please check my last update.
And regarding the color, I'm pretty sure that my implementation is the same as the design.
I just copied the icon as SVG from Figma.
And could you let me know your problem about the fade? If you give me a screenshot, it will be good.

**Comment by predrag-us** [comment]
@winter-sols 

> I moved the CTA button a bit. 

Don't move it 'a bit', place it as stated in the design. 

I see that you actually positioned the button 12px from the bottom and right, but something is getting in the way, therefore the position isn't right. See this screenshot. The transparent blue is where the button should be located, but as mentioned, something is preventing that:

CleanShot 2023-07-24 at 14 09 42@2x

---

> And regarding the color, I'm pretty sure that my implementation is the same as the design.

The colors are not right in the preview. 

Also, please check the shadows and states: default, hover, and active. 

---

> And could you let me know your problem about the fade? 

Notice that the fade doesn't disappear when I scroll fully to the right.

But, somewhere around 99%, it disappears.

https://share.cleanshot.com/GpbxYKM5

**Comment by winter-sols** [comment]
@predrag-us I can't see the fade issue on my local.

https://github.com/holdex/holdex-venture-studio/assets/100481931/811b9044-e974-429e-b76c-07cfd4177d58

**Comment by winter-sols** [comment]
@predrag-us Except the fade issue, I resolved your problems(position, color).
Please review again.

**Comment by predrag-us** [comment]
Preview update required.
cc: @georgeciubotaru

**Comment by predrag-us** [comment]
@winter-sols 

- The CTA is too small now, it's 43.56px while it should be 56px by design.
- Please notice the shadows on the CTA. Once again, be aware of the states and theme.

@georgeciubotaru Review done.

**Comment by winter-sols** [comment]
@predrag-us Please review again.

**Comment by predrag-us** [comment]
@winter-sols 

Now the button is white on dark mode and has chopped edges on both dark and light modes. 

Please notice: 

CleanShot 2023-07-25 at 15 10 37@2x

**Comment by winter-sols** [comment]
@predrag-us Updated again. You need to review.

**Comment by predrag-us** [comment]
Update the preview, please. Thank you!
@georgeciubotaru @markholdex

**Comment by winter-sols** [comment]
@georgeciubotaru Could you redeploy on Vercel?

**Comment by predrag-us** [comment]
@winter-sols 

That problem is solved. Good job! 👍

---

CleanShot 2023-07-26 at 09 14 18@2x

Can you check this padding on smaller breakpoints? It doesn't match the design. 

Also, the secondary navigation doesn't have the same height as in the design. It may be because the padding on `NavItem` has been reduced from 16px to 8px.

Please recheck all paddings and sizes once again. Thank you.

**Comment by winter-sols** [comment]
@predrag-us please review again.

**Comment by predrag-us** [comment]
@georgeciubotaru Get we get a new preview, please? Thanks!

**Comment by winter-sols** [comment]
@georgeciubotaru @predrag-us 
Any updates?

**Comment by predrag-us** [comment]
@winter-sols 

The primary navigation is alright, but there's still something wrong with the secondary navigation. 

Here's a tiny bit of help to speed things up:
- Secondary navigation, update top and bottom padding from 16px to 8px, mobile breakpoints. 
- Secondary navigation, reduce the gap between links and icons from 32px to 24px, mobile breakpoints.
- Secondary navigation, check the color of the icons on both color modes, all breakpoints.
- Secondary navigation, check the position of the underline (screenshot, it's not laying on the line), all breakpoints.

CleanShot 2023-07-26 at 14 33 20@2x

**Comment by winter-sols** [comment]
@predrag-us Okay, let me fix them

**Comment by winter-sols** [comment]
@predrag-us Updated again.

**Comment by winter-sols** [comment]
@predrag-us Could you check it and give me updates?

**Comment by predrag-us** [comment]
@winter-sols 

You've fixed all the issues I addressed but created a new one. 

Recheck the gap between the icons. It's 32px instead of 16px.
Recheck the gap between the `NavItems, it's 32px instead of 24px.

If the next preview doesn't emerge with a new problem, we can finally wrap this up.

**Comment by winter-sols** [comment]
@predrag-us Updated again.

**Comment by predrag-us** [comment]
@georgeciubotaru Please trigger the new preview when you get a chance. Thank you!

**Comment by winter-sols** [comment]
@predrag-us any updates?

**Comment by georgeciubotaru** [comment]
@predrag-us Provided

**Comment by winter-sols** [comment]
@georgeciubotaru @predrag-us All good?

**Comment by predrag-us** [comment]
Finally, all is good on the design side!
@winter-sols @georgeciubotaru

File: src/routes/(pages)/+layout.svelte [hunk 2]
```diff
@@ -38,6 +39,49 @@
...
+      var st = window.scrollY || document.documentElement.scrollTop;
+      if (st > lastScrollTop) {
+          status = false
+      } else if (st < lastScrollTop) {
+          status = true
+      }
+
+      lastScrollTop = st <= 0 ? 0 : st;
+    }, false);
+  });
```
Comment (georgeciubotaru) [inline src/routes/(pages)/+layout.svelte:59]: svelte:window see

Comment (winter-sols) [reply src/routes/(pages)/+layout.svelte:59]: I tried to use `<svelte:window bind:scrollY />` with `svelte-previous` to implement `wherever you are on the page, on scroll up the navigation should reveal`, but it didn't work well.

**Comment by winter-sols** [comment]
@predrag-us @georgeciubotaru 
Any updates?

File: src/lib/components/Button/CTAIcon.svelte [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+			</g>
+		</svg>
+	{:else if theme === "moon"}
+		<svg width="56" height="56" viewBox="8 4 56 56" fill="#2E83E5">
+			<g>
+				<rect x="8" y="4" width="56" height="56" rx="12" fill="#2E83E5"/>
+				<path d="M23 33.0133C23 35.1467 24.4973 37.0053 26.6093 37.316C28.0333 37.5253 29.4733 37.688 30.9267 37.8013C31.548 37.8507 32.1173 38.176 32.464 38.696L36 44L39.536 38.696C39.8827 38.176 40.452 37.8507 41.0733 37.8027C42.5267 37.688 43.9667 37.5253 45.3907 37.316C47.5027 37.0053 49 35.148 49 33.012V24.988C49 22.852 47.5027 20.9947 45.3907 20.684C42.2813 20.2276 39.1427 19.999 36 20C32.8107 20 29.6747 20.2333 26.6093 20.684C24.4973 20.9947 23 22.8533 23 24.988V33.012V33.0133Z" stroke="#FFF" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"/>
+			</g>
+		</svg>
```
Comment (georgeciubotaru) [inline src/lib/components/Button/CTAIcon.svelte:27]: we Should use the icon from `$components/Icon` file (see the name in Figma)

File: src/routes/(pages)/+layout.svelte [hunk 4]
```diff
@@ -38,6 +39,49 @@
...
+
+  onMount(() => {
```
Comment (georgeciubotaru) [inline src/routes/(pages)/+layout.svelte:61]: define an action better

File: src/routes/(pages)/layout.pug [hunk 5]
```diff
@@ -39,28 +39,35 @@ mixin mobile-nav
 
 .app-container
   header(
-    class!="z-10 w-full block  relative {scrollY > 1 ? 'sticky top-0 border-b border-solid border-l4 bg-l1' : ''}"
+    class!="z-10 w-full block relative sticky top-0"
   )
     .outer-container(
-      class="relative xs:before:content['] xs:before:absolute xs:before:left-0 xs:before:w-16 xs:before:h-16 xs:before:z-10 xs:before:bg-gradient-to-r xs:after:bg-gradient-to-l from-l1 to-transparent xs:after:content['] xs:after:absolute xs:after:right-0 xs:after:w-16 xs:after:h-16 xs:after:z-10 xs:after:top-0"
+      class=""
```
Comment (georgeciubotaru) [inline src/routes/(pages)/layout.pug:45]: empty class declaration

File: src/lib/components/Button/CTAIcon.svelte (hunk 3 above)
Comment (winter-sols) [reply src/lib/components/Button/CTAIcon.svelte:27]: Could you let me know the name? I can't find that in Figma.

**Comment by winter-sols** [comment]
@georgeciubotaru I fixed your comments.

**Comment by markholdex** [comment]
@predrag-us please have a look.

**Comment by predrag-us** [comment]
@winter-sols 

CTA shouldn't be visible on large screens.

CTA doesn't work, nothing happens on click.
Also, the whole CTA should be clickable, not just the icon.

**Comment by winter-sols** [comment]
@predrag-us Let me check

**Comment by winter-sols** [comment]
@georgeciubotaru I updated. need to deploy

**Comment by predrag-us** [comment]
@winter-sols All good. 💪

**Comment by winter-sols** [comment]
@predrag-us Perfect!
Could you approve it so that it can be merged?

**Comment by predrag-us** [comment]
@georgeciubotaru Please check the code once again and approve / merge if everything is okay.

File: src/routes/(pages)/+layout.svelte [hunk 6]
```diff
@@ -38,6 +38,47 @@
     localStorage.setItem('theme', themeIconName === 'moon' ? 'light' : 'dark');
   };
 
+  var lastScrollTop = 0;
+  var secondaryNavScrollLeft = 0
+  var status = true
+  var isLeftEnd = true
+  var isRightEnd = false
+
```
Comment (georgeciubotaru) [inline src/routes/(pages)/+layout.svelte:46]: `let` or `const`

**Comment by winter-sols** [comment]
@georgeciubotaru fixed!

**Comment by winter-sols** [comment]
@georgeciubotaru anything else?

File: src/routes/(pages)/+layout.svelte (hunk 1 above)
Comment (georgeciubotaru) [reply src/routes/(pages)/+layout.svelte:174]: @winter-sols comment these styles

**Comment by winter-sols** [comment]
@georgeciubotaru Could you make some more time for this issue? I have worked this ticket for over a week.

**Review by georgeciubotaru — APPROVED** [review]

**Pull request merged**